From: Bruce Allan <bruce.w.allan@intel.com>
Date: Thu, 9 Dec 2010 23:04:25 +0000
Subject: e1000e: static analysis tools complain of a possible null ptr p dereference
References: bnc#699092, fate#311825
Patch-mainline: v2.6.38
Git-commit: 61c758166701c4f156c0aba10260f5729b1a6f43

Adding this default case resolves the issue.

v2- Removed "break" in default case based on feedback

Signed-off-by: Bruce Allan <bruce.w.allan@intel.com>
Tested-by: Emil Tantilov <emil.s.tantilov@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Acked-by: Cho, Yu-Chen <acho@suse.com>
---
 drivers/net/e1000e/ethtool.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/net/e1000e/ethtool.c
+++ b/drivers/net/e1000e/ethtool.c
@@ -1990,6 +1990,9 @@ static void e1000_get_ethtool_stats(stru
 			p = (char *) adapter +
 					e1000_gstrings_stats[i].stat_offset;
 			break;
+		default:
+			data[i] = 0;
+			continue;
 		}
 
 		data[i] = (e1000_gstrings_stats[i].sizeof_stat ==
